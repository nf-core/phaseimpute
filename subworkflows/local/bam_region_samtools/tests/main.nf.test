nextflow_workflow {

    name "Test Subworkflow BAM_REGION_SAMTOOLS"
    script "../main.nf"
    config "./nextflow.config"

    workflow "BAM_REGION_SAMTOOLS"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "subworkflows/bam_region_samtools"
    tag "bam_region_samtools"

    tag "samtools"
    tag "samtools_view"
    tag "samtools_merge"
    tag "samtools_index"

    test("Get subregion of bam file 10000bp") {
        when {
            params {
                max_cpus   = 2
                max_memory = '2.GB'
            }
            workflow {
                """
                input[0] = input[0] = Channel.fromList([
                    [
                        [id: "NA12878"],
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA12878/NA12878.s.bam", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA12878/NA12878.s.bam.bai", checkIfExist:true),
                    ],
                    [
                        [id: "NA19401"],
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA19401/NA19401.s.bam", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA19401/NA19401.s.bam.bai", checkIfExist:true),
                    ],
                ])
                input[1] = Channel.of([
                    [chr: "chr22", region: "chr22:16600000-16610000"], "chr22:16600000-16610000"
                ]).collect()
                input[2] = Channel.of([
                    [id: "GRCh38"],
                    file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38.s.fa.gz", checkIfExist:true),
                    file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38.s.fa.gz.fai", checkIfExist:true),
                ]).collect()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert snapshot(workflow.out.bam_region.collect{ bam(it[1]).getReads().size() }).match("Summary10000bp") }
            )
        }
    }

    test("Get subregion of bam file 1000bp") {
        when {
            params {
                max_cpus   = 2
                max_memory = '2.GB'
            }
            workflow {
                """
                input[0] = input[0] = Channel.fromList([
                    [
                        [id: "NA12878"],
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA12878/NA12878.s.bam", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA12878/NA12878.s.bam.bai", checkIfExist:true),
                    ],
                    [
                        [id: "NA19401"],
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA19401/NA19401.s.bam", checkIfExist:true),
                        file(params.pipelines_testdata_base_path + "hum_data/individuals/NA19401/NA19401.s.bam.bai", checkIfExist:true),
                    ],
                ])
                input[1] = Channel.of([
                    [chr: "chr22", region: "chr22:16609000-16610000"], "chr22:16609000-16610000"
                ]).collect()
                input[2] = Channel.of([
                    [id: "GRCh38"],
                    file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38.s.fa.gz", checkIfExist:true),
                    file(params.pipelines_testdata_base_path + "hum_data/reference_genome/GRCh38.s.fa.gz.fai", checkIfExist:true),
                ]).collect()
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
                { assert snapshot(workflow.out.bam_region.collect{ bam(it[1]).getReads().size() }).match("Summary1000bp") }
            )
        }
    }
}
