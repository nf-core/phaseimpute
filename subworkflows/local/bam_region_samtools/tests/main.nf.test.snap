{
    "Get subregion of bam file 10000bp": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "NA12878",
                            "chr": "all"
                        },
                        "NA12878.bam:md5,382b24f44346ec2027e8578a5085c20f",
                        "NA12878.bam.bai:md5,484505b0727206e1e523dd90fa31cd16"
                    ],
                    [
                        {
                            "id": "NA19401",
                            "chr": "all"
                        },
                        "NA19401.bam:md5,9efa18406f24954c75934d928ed9c062",
                        "NA19401.bam.bai:md5,5a9429b877ffee92f7694d3579b4354b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,466438b58e39f6156e98d049f38e9d71",
                    "versions.yml:md5,7c6c94bfec70fe9da9e00f94b45f38f5",
                    "versions.yml:md5,966689bbca491e03935dd21220e66a1e"
                ],
                "bam_region": [
                    [
                        {
                            "id": "NA12878",
                            "chr": "all"
                        },
                        "NA12878.bam:md5,382b24f44346ec2027e8578a5085c20f",
                        "NA12878.bam.bai:md5,484505b0727206e1e523dd90fa31cd16"
                    ],
                    [
                        {
                            "id": "NA19401",
                            "chr": "all"
                        },
                        "NA19401.bam:md5,9efa18406f24954c75934d928ed9c062",
                        "NA19401.bam.bai:md5,5a9429b877ffee92f7694d3579b4354b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,466438b58e39f6156e98d049f38e9d71",
                    "versions.yml:md5,7c6c94bfec70fe9da9e00f94b45f38f5",
                    "versions.yml:md5,966689bbca491e03935dd21220e66a1e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-17T16:44:44.476760014"
    },
    "Summary10000bp": {
        "content": [
            [
                2498,
                2510
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-17T16:44:44.886794567"
    },
    "Get subregion of bam file 1000bp": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "NA12878",
                            "chr": "all"
                        },
                        "NA12878.bam:md5,bb7e4faf368fa9c3bce66db3632fe05b",
                        "NA12878.bam.bai:md5,8d864fd88a20d14d4aff3de607655cbb"
                    ],
                    [
                        {
                            "id": "NA19401",
                            "chr": "all"
                        },
                        "NA19401.bam:md5,159d5b70d052e1269698635e28a8cbc9",
                        "NA19401.bam.bai:md5,ee992cd617b553c5fcfddd897193a0e5"
                    ]
                ],
                "1": [
                    "versions.yml:md5,466438b58e39f6156e98d049f38e9d71",
                    "versions.yml:md5,7c6c94bfec70fe9da9e00f94b45f38f5",
                    "versions.yml:md5,966689bbca491e03935dd21220e66a1e"
                ],
                "bam_region": [
                    [
                        {
                            "id": "NA12878",
                            "chr": "all"
                        },
                        "NA12878.bam:md5,bb7e4faf368fa9c3bce66db3632fe05b",
                        "NA12878.bam.bai:md5,8d864fd88a20d14d4aff3de607655cbb"
                    ],
                    [
                        {
                            "id": "NA19401",
                            "chr": "all"
                        },
                        "NA19401.bam:md5,159d5b70d052e1269698635e28a8cbc9",
                        "NA19401.bam.bai:md5,ee992cd617b553c5fcfddd897193a0e5"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,466438b58e39f6156e98d049f38e9d71",
                    "versions.yml:md5,7c6c94bfec70fe9da9e00f94b45f38f5",
                    "versions.yml:md5,966689bbca491e03935dd21220e66a1e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-17T16:45:00.483599493"
    },
    "Summary1000bp": {
        "content": [
            [
                368,
                327
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-17T16:45:00.954762728"
    }
}